packages/actor-http-native/lib/ActorHttpNative.ts

```diff
@@ -63,41 +63,56 @@ export class ActorHttpNative extends ActorHttp {
       const req = this.requester.createRequest(options);
       req.on('error', reject);
       req.on('response', httpResponse => {
-        httpResponse.on('error', (error: Error) => {
-          httpResponse = null;
-          reject(error);
-        });
-        // Avoid memory leak on HEAD requests
-        if (options.method === 'HEAD') {
-          httpResponse.destroy();
-        }
-        // Using setImmediate so error can be caught should it be thrown
-        setImmediate(() => {
-          if (httpResponse) {
-            // Expose fetch cancel promise
-            httpResponse.cancel = () => {
-              httpResponse.destroy();
-              return Promise.resolve(undefined);
-            };
-            // Missing several of the required fetch fields
-            const headers = new Headers(httpResponse.headers);
-            const result = <IActorHttpOutput> {
-              body: httpResponse,
-              headers,
-              ok: httpResponse.statusCode < 300,
-              redirected: options.url !== httpResponse.responseUrl,
-              status: httpResponse.statusCode,
-              // When the content came from another resource because of conneg, let Content-Location deliver the url
-              url: headers.has('content-location') ? headers.get('content-location') : httpResponse.responseUrl,
-            };
-            resolve(result);
-          }
-        });
+        processHttpResponse(httpResponse, options, resolve, reject);
       });
     });
   }
 }
 
+export function processHttpResponse(httpResponse: any, options: any, resolve: any, reject: any): void {
+  httpResponse.on('error', (error: Error) => {
+    httpResponse = null;
+    reject(error);
+  });
+  // Avoid memory leak on HEAD requests
+  if (options.method === 'HEAD') {
+    httpResponse.destroy();
+  }
+  // Using setImmediate so error can be caught should it be thrown
+  setImmediate(() => {
+    if (httpResponse) {
+      // Expose fetch cancel promise
+      httpResponse.cancel = () => {
+        httpResponse.destroy();
+        return Promise.resolve(undefined);
+      };
+      // Missing several of the required fetch fields
+
+      let headers = new Headers(httpResponse.headers);
+      if (httpResponse && httpResponse.input && httpResponse.input.headers) {
+        headers = httpResponse.input.headers;
```

Comment:
Why is this needed here? Is the `httpResponse.headers` not sufficient?
Not sure, but I would think `httpResponse.input.headers` refers to the original request headers, which should not be in here, right?